### DISCUSSION TITLE
Version 4.0 features, changes, etc

### DISCUSSION THREAD

**Discussion by gitbrent** [gitbrent/PptxGenJS#1070] [Ideas]
# 👋 Welcome!

I've got some ideas for what's coming in the [4.0 Release](https://github.com/gitbrent/PptxGenJS/milestone/37), so follow along or chime in and let's build together 💪.

## ✅  Changes: Major Features

- [ ] Add support for connected shapes 
  - Pull #1059
  - Pull #777
- [ ] Add support for groups
  - Pull #929 (Issue #307)

## ✅  Changes: Minor Features

- [ ] TextBaseProps.bullet - remove boolean

## ⚠️ Changes: Deprecated/Breaking

- [ ] Units: `TextPropsOptions` `margin` prop is points but PowerPoint uses inches 
- [ ] Naming: A few exported enums in `core-enums.ts` file are MixedCase while most are uppercase (e.g.: `CHART_TYPE`)
- [ ] Naming: rename `SchemeColor` to `ThemeColor`
- [ ] Change core chart props to be less vague
```javascript
// OLD
{
  name: 'Location',
  labels: ['West', 'East'],
  values: [10, 20],
  sizes: [5, 3],
}
// NEW
{
  serName: 'Location',
  catLabels: ['West', 'East'],
  catValues: [10, 20],
  catBubbleSizes: [5, 3],
}
```
- [ ] Change `slide` methods to use named arguments instead of positional
```javascript
// OLD
export interface PresSlide extends SlideBaseProps {
    addChart: (type: CHART_NAME | IChartMulti[], data: any[], options?: IChartOpts) => PresSlide
    addImage: (options: ImageProps) => PresSlide
    addMedia: (options: MediaProps) => PresSlide
    addNotes: (notes: string) => PresSlide
    addShape: (shapeName: SHAPE_NAME, options?: ShapeProps) => PresSlide
    addTable: (tableRows: TableRow[], options?: TableProps) => PresSlide
    addText: (text: string | TextProps[], options?: TextPropsOptions) => PresSlide
}

// NEW
export interface IChartArgs {
  type: ..., 
  data: ..., 
  props?: IChartProps
}
export interface PresSlide extends SlideBaseProps {
    addChart: (IChartArgs) => PresSlide
}
```
## ♻️ Changes: Internal/Code

- Remove all `@deprecated` props from codebase (e.g.: `props.bkgd`)
- remove "@deprecated" props from `core-interfaces.ts` and `index.d.ts` (eg: `alpha?: number`)

## 🤔 Questions

- [_what do non-English countries and/or metric countries use if not inches?_]

**Comment by AprilJer** [comment]
Hi @gitbrent,

We are looking forward of the `group` feature in V4.0. I am wondering do you have a rough timeline on when it would be available?

Thanks a lot for your awesome work!

**Reply by gitbrent** [reply]
Q3-Q4 of 2022

**Reply by zileeva** [reply]
Hello! Any updates on when the release is coming? 
Thank you.

**Comment by redfoxlee** [comment]
Hi @gitbrent 

> anything or only major?

I don't know if this should be a major or minor, but personally i would like to have the Comment feature, is this possible?

> what do non-English countries and/or metric countries use if not inches?

The metric system. Like centimeters. I'm not so sure...

Thanks a lot for your awesome work!

**Comment by twimsatt** [comment]
I would like the rotate option to work in table header fields. I have a discussion topic started for this. It works for text but not for table headers even when embedded in an array of text objects.

**Comment by zileeva** [comment]
Hello @gitbrent ,

We are looking forward to V4.0 release and especially to the `group` feature. When do you plan to make this available?

Thank you!

Best,
Yulia

**Comment by MarcoLasticot** [comment]
Hi @gitbrent 

Do you have any news on v4.0 ? Group feature is really waited.
Thank you

**Comment by dadakurt** [comment]
Hello @gitbrent,

Do you have any idea if version 4 will be available soon?
Everybody wait to group the shapes :)

Thanks a lot for your feedback !

**Comment by sambauers** [comment]
> [what do non-English countries and/or metric countries use if not inches?]

In Australia, PowerPoint on Mac will set units to Centimetres.

**Comment by MarcoLasticot** [comment]
Does anyone know if this project is still active ?
If not, do you know some alternative package ?
Thanks

**Comment by shenggen1987** [comment]
@gitbrent    Support custom property, you can preview： https://github.com/gitbrent/PptxGenJS/issues/1417

**Comment by giona69** [comment]
is there anyone to contact for the update of this project? I solved an issue and we need to post a PR; also there are many PRs that are read and working, can anyone merge them? If none is managing the repo is possible to take over?
